### ISSUE TITLE
Promote TEP-0029 - Isolated Step and Sidecar Workspaces

### ISSUE THREAD

**Issue by pritidesai** [tektoncd/pipeline#6109]
### Current Feature Version

`enable-api-fields: alpha`

### Raise Feature Version

`enable-api-fields: beta`

### Pipeline Release

[Tekton Pipelines 0.24.0](https://github.com/tektoncd/pipeline/releases/tag/v0.24.0)

### Issues Reported by the Users

None

### Usage in Dogfooding Cluster

None

### Examples/Tests

* [TEP-0029](https://github.com/tektoncd/community/blob/main/teps/0029-step-workspaces.md)
* [Implementation PR in Pipelines Repo](https://github.com/tektoncd/pipeline/pull/3700)
* [examples/v1beta1/pipelineruns/alpha/isolated-workspaces.yaml](https://github.com/tektoncd/pipeline/blob/main/examples/v1beta1/pipelineruns/alpha/isolated-workspaces.yaml)
* [examples/v1beta1/taskruns/alpha/authenticating-git-commands.yaml](https://github.com/tektoncd/pipeline/blob/main/examples/v1beta1/taskruns/alpha/authenticating-git-commands.yaml)
* [examples/v1beta1/taskruns/alpha/workspace-in-sidecar.yaml](https://github.com/tektoncd/pipeline/blob/main/examples/v1beta1/taskruns/alpha/workspace-in-sidecar.yaml)
* [examples/v1beta1/taskruns/alpha/workspace-isolation.yaml](https://github.com/tektoncd/pipeline/blob/main/examples/v1beta1/taskruns/alpha/workspace-isolation.yaml)

**Comment by JeromeJu** [comment]
pipeline WG: no end user cases, maybe we could lower the priority of this

**Comment by afrittoli** [comment]
I started looking into this. 
The question I have is how to deal with the `v1beta1` API. 
Until [TEP-0138](https://github.com/tektoncd/community/pull/1034) is implemented, the `v1beta1` API should get beta features on by default, so I think that's what I should implement here for consistency. @lbernick @JeromeJu @pritidesai @vdemeester wdyt?

**Comment by lbernick** [comment]
I agree; we should have consistency with our existing beta features (resolvers, object params etc).
